From 6faaf6e8dad1418d17c0cb689861c2678fcbdd75 Mon Sep 17 00:00:00 2001
From: Milad Fa <mfarazma@redhat.com>
Date: Tue, 16 Nov 2021 12:30:39 -0500
Subject: [PATCH] PPC/s390: Activate Argument Count Consistency

Port ea6dd0f4f622f8d0cc4a446e348668d1a5f329c3

Original Commit Message:

    Activate argument count consistency (receiver is always included in
    JS argument count) for arm and arm64.

R=pthier@chromium.org, joransiu@ca.ibm.com, junyan@redhat.com, midawson@redhat.com
BUG=
LOG=N

Change-Id: Ib5fe912dfc108e95af6eab46d5a7672ac209e652
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3285404
Reviewed-by: Patrick Thier <pthier@chromium.org>
Reviewed-by: Junliang Yan <junyan@redhat.com>
Commit-Queue: Milad Fa <mfarazma@redhat.com>
Cr-Commit-Position: refs/heads/main@{#77933}
---
 BUILD.gn | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/BUILD.gn b/BUILD.gn
index 36cf51ba9a..c7d5cbdfa5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -357,7 +357,8 @@ declare_args() {
       v8_current_cpu == "x86" || v8_current_cpu == "x64" ||
       v8_current_cpu == "arm" || v8_current_cpu == "arm64" ||
       v8_current_cpu == "mips64el" || v8_current_cpu == "mipsel" ||
-      v8_current_cpu == "loong64" || v8_current_cpu == "riscv64"
+      v8_current_cpu == "loong64" || v8_current_cpu == "riscv64" ||
+      v8_current_cpu == "ppc64" || v8_current_cpu == "s390x"
 }
 
 # Derived defaults.
-- 
2.35.1

